Staging/2111 release notes clean (#1232)
o3de/o3de.org: content/docs/release-notes/archive/21-11-release-notes.md

---
linktitle: 2111.1 Release Notes
title: Release Notes for Open 3D Engine 2111.1
description: Find out what was delivered in O3DE 2111.1
weight: 101
toc: true
---

> Good news, everyone! The Open 3D Foundation is proud to announce that we've delivered our first stable release of Open 3D Engine (O3DE), version 21.11.

Reviewer: Shouldn't the version number in this first paragraph match the title?
```suggestion
Good news, everyone! The Open 3D Foundation is proud to announce that we've delivered our first stable release of Open 3D Engine (O3DE), version 2111.1.
```

## Highlights

> ### Open 3D Engine Binaries Installer

Reviewer: ```suggestion
### Open 3D Engine binaries installer
```

There's now a binary installer for Windows for Open 3D Engine! Now you can try out our creative tools without building from source every time. This model - sometimes called "Open 3D Engine as an SDK" or "Open 3D Engine toolkit" - delivers pre-built binaries, DLLs, and headers which you can use for local development. For users who want to try out our tools without going through build time iterations, or are only interested in seeing how to write Gems and extensions to O3DE which don't require core source changes, the installer is an ideal new way to try out these workflows and find issues with us.

For everyone who's joining us as a creative worker in simulations and games eager to try out Open 3D Engine: Please keep in mind our journey is just beginning! Don't expect to make a full, production-ready game in O3DE - yet. Right now we want our community to grow by playing with these tools and giving us feedback that we need to make O3DE a long-term success. That means if you're committed to trying the binary installer, be prepared for hiccups as you work, and be ready to file bugs (or maybe even make a code contribution!) as you encounter issues.

> Want to try it? [Download the Windows installer for O3DE 21.11 Stable!](https://o3debinaries.org/download/windows.html)
> 
> {{< note >}}
> For this release of O3DE, we're using two version numbers due to some technical limitations of Windows that we're working on resolving. The source releases of O3DE are versioned as **2111.1**, and binary installers are versioned as **21.11 (Stable)**.
> {{< /note >}} 

Reviewer 2: ```suggestion
Want to try it? [Download the Windows installer for O3DE Stable 21.11!](https://o3debinaries.org/download/windows.html)

{{< note >}}
For this release of O3DE, we're using two version numbers due to some technical limitations of Windows that we're working on resolving. The source releases of O3DE are versioned **2111.1**, and binary installers are named **Stable 21.11**.
{{< /note >}} 
```

### Linux support

Open 3D Engine is now considered to be available in preview for Linux! Don't expect to have support for everything you need quite yet, and we're still making sure that all the bugs and issues are ironed out, but you can now run O3DE client applications and runtimes - like the O3DE Editor itself - on Linux desktop.

![The O3DE Editor running on an X desktop environment on Ubuntu 20.04 LTS (Focal Fossa)](/images/release-notes/2111-1/linux-desktop.png)

> Right now we only have official support for Debian 20.04 LTS (Focal Fossa) as a pre-built binary distribution, but many Linux systems which meet the [hardware and software requirements](/docs/welcome-guide/requirements/) can be modified to successfully run Open 3D Engine. To get started building O3DE and your projects on Linux, check out the [Linux getting started documentation](docs/welcome-guide/setup/setup-from-github/building-linux/)

Reviewer: ```suggestion
Right now we only have official support for Debian 20.04 LTS (Focal Fossa) as a pre-built binary distribution, but many Linux systems which meet the [hardware and software requirements](/docs/welcome-guide/requirements/) can be modified to successfully run Open 3D Engine. To get started building O3DE and your projects on Linux, check out the [Linux getting started documentation](/docs/welcome-guide/setup/setup-from-github/building-linux/)
```

Reviewer 3: ``Debian 20.04 LTS`` is not correct. Ubuntu is based on Debian, so they are not one in the same. Version 20.04 LTS refers to a specific Ubuntu release. Ubuntu uses debian-based packaging.
```suggestion
Right now we only have official support for Ubuntu 20.04.3 LTS (Focal Fossa) as a pre-built binary debian package. Other Linux systems which meet the [hardware and software requirements](/docs/welcome-guide/requirements/) is considered experimental and may need modifications to successfully run Open 3D Engine. To get started building O3DE and your projects on Linux, check out the [Linux getting started documentation](docs/welcome-guide/setup/setup-from-github/building-linux/)
```

Reviewer: Updated spham's comment to incorporate additional fixes and rewording to mention both choices for getting started - installer and build docs.

> And of course, you can [download the Debian package!](https://o3debinaries.org/download/linux.html)

Reviewer 2: ```suggestion
And of course, you can [download the Debian 20.04 LTS package!](https://o3debinaries.org/download/linux.html)
```

Reviewer: With a suggested link to the installer docs in the previous paragraph, I don't think we even need this sentence any longer.

For content creators, Atom now has support for [TressFX 4.1]( https://github.com/GPUOpen-Effects/TressFX/) via the AtomTressFX Gem. With the AtomTressFX Gem, you have the power to create, model, and simulate realistic hair, fur, and other dense thin-volumed surfaces.

### Editor improvements

Open 3D Engine Editor 2111.1 has major improvements across the board in terms of performance and stability, like the rest of the engine, and content creation workflows are getting a lot of attention! Thanks to user feedback and sessions conducted with the O3DE UI/UX Special Interest Group, we collected a number of important adjustments our community wanted.

Most importantly, Prefabs are now represented as a single object in the world, and you must use the **Edit Prefab** action on a Prefab in the Editor in order to edit the prefab itself. Prefab editing is available through **double-clicking** on an object in the Editor. While locked to the prefab edit mode, other instances of the prefab visible through the Editor viewport will receive real-time updates.

> And another change that we thought was so important that it deserves its own callout: This means is no more "sticky-select" in the Editor by default! You no longer have to double-click when deselecting an entity that you're editing. If you'd like to re-enable this feature, it can be changed in Editor Settings or with the `/Amazon/Preferences/Editor/StickySelect` registry setting.

Reviewer: The first sentence is awkward. Either a key word was dropped after the ":" or it needs to be reworded. Unfortunately, I don't know what was meant here, so I can't offer a suggestion.

### Heightfield-based terrain

We now have an experimental terrain system! Even basic terrain is important for designing outdoor environments and larger-scale worlds. The terrain system uses [heightmaps](https://en.wikipedia.org/wiki/Heightmap) to define deformations over a geometric plane to create world terrain. Anyone familiar with heightmaps should be able to play around with terrain now, and we encourage all of our users to try this feature when creating outdoor scenes in Open 3D Engine.

![Heightfield Terrain editing in the O3DE Editor](/images/release-notes/2111-1/terrain.png)

For the full information on implementation progress and the planned design, please see [O3DE SIG-Content RFC #4](https://github.com/o3de/sig-content/blob/main/rfcs/rfc-4-terrain-system.md).

> ### Project manager external Gems support

Reviewer: ```suggestion
### External Gems support in Project Manager
```

> Open 3D Engine Project Manager how has improvements which allow the support of Gems which exist outside of the official Gem catalog! This feature comes along with the ability to register a source control repository with the Project Manager, and use this as an external source to pull and install Gems for your project. In addition, we've fixed a critical bug: Project Manager now correctly auto-adds and auto-displays Gem dependencies which affect your project configuration.

Reviewer: ```suggestion
Open 3D Engine Project Manager now has improvements which allow the support of Gems that exist outside of the official Gem catalog! This feature comes along with the ability to register a source control repository with the Project Manager, and use this as an external source to pull and install Gems for your project. In addition, we've fixed a critical bug: Project Manager now correctly auto-adds and auto-displays Gem dependencies which affect your project configuration.
```

> ### Script canvas performance and editing

Reviewer: ```suggestion
### Script Canvas performance and editing
```

> 2111.1 contains plenty of performance improvements, but one which affects all project runtimes (and working in the Editor!) is worth calling out. Script Canvas now uses JSON as its new file format, which gives an average 71% size file reduction. Smaller files means faster loads, too! And not just that, but with the migration away from XML serialization and towards more flexible JSON tools, it's easier than ever to write external tools and scripts that can ingest Script Canvas JSON files.

Reviewer: ```suggestion
2111.1 contains plenty of performance improvements, but one which affects all project runtimes and performance in the Editor is worth calling out. Script Canvas now uses JSON as its file format instead of XML, which gives an average 71% size file reduction. Smaller files means faster loads, too! And not just that, but with the migration away from XML serialization and towards more flexible JSON tools, it's easier than ever to write external tools and scripts that can ingest Script Canvas JSON files.
```
